### ISSUE TITLE
[v6.1] Add support for PHP 8.1

### ISSUE THREAD

**Issue by llaville** [oxsecurity/megalinter#2086]
**Is your feature request related to a problem? Please describe.**
Actually, if you want to use [PHPStan](https://oxsecurity.github.io/megalinter/6.13.0/descriptors/php_phpstan/) linter on a PHP project that required at least PHP 8.1, you got an error as follow 

![phpstan-php8](https://user-images.githubusercontent.com/364342/202657635-12b1d664-5925-46fc-bfcd-b539ea50b420.png)

Chunk of **.mega-linter.yml**
```yaml
ENABLE:
    - ACTION
    - EDITORCONFIG
    - JSON
    - MARKDOWN
    - PHP
    - YAML
DISABLE_LINTERS:
    - JSON_JSONLINT
    - JSON_V8R
    - MARKDOWN_MARKDOWN_TABLE_FORMATTER
    - MARKDOWN_MARKDOWN_LINK_CHECK
    - PHP_BUILTIN
    - PHP_PSALM
    - YAML_V8R
PHP_PHPSTAN_CLI_LINT_MODE: project
PHP_PHPLINT_CLI_LINT_MODE: project
PHP_PHPCS_CLI_LINT_MODE: project
PHP_PHPSTAN_PRE_COMMANDS:
    - command: "update-alternatives --set php /usr/bin/php8"
      cwd: root

```

**Describe the solution you'd like**
Related to discussion started from https://github.com/oxsecurity/megalinter/pull/2039#issuecomment-1302997006 about drop support of PHP 7, we should consider to add support both PHP 8.0 and 8.1 (and perharps later 8.2.0 will come shortly now in stable status) 

The reason came from PHP descriptors used in MegaLinter v6.1 : https://oxsecurity.github.io/megalinter/latest/descriptors/php/

Especially https://pkgs.alpinelinux.org/packages?branch=edge&name=php8

PHP 8.1 is only supported in Alpine Linux Package since v3.16; see 
- https://pkgs.alpinelinux.org/packages?name=php81&branch=v3.16&repo=&arch=&maintainer=
- https://pkgs.alpinelinux.org/packages?name=php81&branch=edge&repo=&arch=&maintainer=

**Describe alternatives you've considered**
Disable temporay PHPStan linter for project that requires PHP 8.1 or greater

**Comment by nvuillam** [comment]
Do we need two separate installations for php 8.0 et php 8.1 ?
8.0 projects are not compliant with 8.1 , as the semver step is minor ?

**Comment by llaville** [comment]
> 8.0 projects are not compliant with 8.1 , as the semver step is minor ?

Yes of course it's compliant, but there are lot of project in the PHP community that consider to require PHP 8.1 as the minimum requirement.

You will find such Composer constraint 
```json
"require": {
        "php": "^8.1"
}
```

The reasons are I think 
- active support  of PHP 8.0 will stop soon : https://www.php.net/supported-versions.php
- performance improved between PHP 8.0 and 8.1 : https://kinsta.com/fr/blog/comparaison-php/  (in french, sorry for english audience)

**Comment by nvuillam** [comment]
Ok so let's remove php7 support and install PHP8 by default ? ^^

**Comment by llaville** [comment]
PHP 7 security support will be dropped very soon !
So it's a frame time to prepare a MegaLinter v6 with both PHP 8.0 (default) and PHP 8.1 support. 

Or perharps PHP 8.0 and PHP 8.1 (default) support. 

I suggest to use this kind or priority to match PHP versions

```
update-alternatives --install /usr/bin/php php /usr/bin/php8 100
update-alternatives --install /usr/bin/php php /usr/bin/php81 110
```
And perharps later when Alpine version will be available with upcoming PHP 8.2
```
update-alternatives --install /usr/bin/php php /usr/bin/php82 120
```

**Comment by nvuillam** [comment]
As far as i know the only thing blocking to use latest alpine is ...php7 support, so the planets seem to be aligned ^^

**Comment by llaville** [comment]
The best solution won't be to install both PHP 8.0 and 8.1 ; only PHP 8.1 is necessary because it support also PHP 8.0 ;-)

**Comment by llaville** [comment]
@nvuillam Hello

While I'm trying to prepare a PR to drop PHP 7 and replace by PHP 8.1, I'm in trouble with MegaLinter v6 (commit `5cbf1b48f` into main branch)

I've followed [Contributing Guide](https://megalinter.io/latest/contributing/)


With basic requirements : 

* **Python 3.7.5**
* **Node v14.17.5**


Run `. .venv/bin/activate`

Then `echo ".venv/" >> .git/info/exclude`

Then `python3 -m pip install -U pip`

Getting this output:
```
Cache entry deserialization failed, entry ignored
Collecting pip
  Downloading https://files.pythonhosted.org/packages/09/bd/2410905c76ee14c62baf69e3f4aa780226c1bbfc9485731ad018e35b0cb5/pip-22.3.1-py3-none-any.whl (2.1MB)
    100% |████████████████████████████████| 2.1MB 315kB/s
Installing collected packages: pip
  Found existing installation: pip 9.0.1
    Uninstalling pip-9.0.1:
      Successfully uninstalled pip-9.0.1
Successfully installed pip-22.3.1
(.venv) llaville@DESKTOP-83207V7:~/devilbox_data/backups/github/megalinter-oxsec$ python3 -m pip install -r requirements.dev.txt
Collecting gitpython
  Downloading GitPython-3.1.29-py3-none-any.whl (182 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 182.5/182.5 kB 1.1 MB/s eta 0:00:00
Collecting jsonschema
  Downloading jsonschema-4.17.3-py3-none-any.whl (90 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.4/90.4 kB 1.6 MB/s eta 0:00:00
Collecting jsonpickle
  Downloading jsonpickle-3.0.1-py2.py3-none-any.whl (40 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.5/40.5 kB 396.6 kB/s eta 0:00:00
Collecting mkdocs
  Downloading mkdocs-1.4.2-py3-none-any.whl (3.7 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.7/3.7 MB 1.0 MB/s eta 0:00:00
Collecting multiprocessing_logging
  Downloading multiprocessing_logging-0.3.3-py2.py3-none-any.whl (8.3 kB)
Collecting pychalk
  Using cached pychalk-2.0.1.tar.gz (17 kB)
  Preparing metadata (setup.py) ... done
Collecting pygithub
  Downloading PyGithub-1.57-py3-none-any.whl (297 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 297.6/297.6 kB 582.0 kB/s eta 0:00:00
Collecting commentjson
  Using cached commentjson-0.9.0.tar.gz (8.7 kB)
  Preparing metadata (setup.py) ... done
Collecting pytablewriter
  Downloading pytablewriter-0.64.2-py3-none-any.whl (106 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 106.6/106.6 kB 762.7 kB/s eta 0:00:00
Collecting pytest-cov
  Downloading pytest_cov-4.0.0-py3-none-any.whl (21 kB)
Collecting pytest-timeout
  Downloading pytest_timeout-2.1.0-py3-none-any.whl (12 kB)
Collecting pytest-xdist
  Downloading pytest_xdist-3.1.0-py3-none-any.whl (36 kB)
Collecting python-gitlab
  Downloading python_gitlab-3.12.0-py3-none-any.whl (130 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 131.0/131.0 kB 840.2 kB/s eta 0:00:00
Collecting pyyaml
  Using cached PyYAML-6.0-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (596 kB)
Collecting requests
  Downloading requests-2.28.1-py3-none-any.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.8/62.8 kB 291.8 kB/s eta 0:00:00
Collecting terminaltables
  Downloading terminaltables-3.1.10-py2.py3-none-any.whl (15 kB)
Collecting webpreview
  Downloading webpreview-1.7.2-py3-none-any.whl (18 kB)
Collecting yq
  Downloading yq-3.1.0-py3-none-any.whl (17 kB)
Collecting importlib-metadata>=3.10
  Downloading importlib_metadata-5.1.0-py3-none-any.whl (21 kB)
Collecting mkdocs-material
  Downloading mkdocs_material-8.5.11-py3-none-any.whl (7.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.6/7.6 MB 442.5 kB/s eta 0:00:00
Collecting mdx_truly_sane_lists
  Downloading mdx_truly_sane_lists-1.3-py3-none-any.whl (6.1 kB)
Collecting beautifulsoup4
  Downloading beautifulsoup4-4.11.1-py3-none-any.whl (128 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 128.2/128.2 kB 400.3 kB/s eta 0:00:00
Collecting giturlparse
  Using cached giturlparse-0.10.0-py2.py3-none-any.whl (14 kB)
Collecting json-schema-for-humans
  Downloading json_schema_for_humans-0.44-py3-none-any.whl (70 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 70.5/70.5 kB 341.1 kB/s eta 0:00:00
Collecting gitdb<5,>=4.0.1
  Downloading gitdb-4.0.10-py3-none-any.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.7/62.7 kB 556.1 kB/s eta 0:00:00
Collecting typing-extensions>=3.7.4.3
  Downloading typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting importlib-resources>=1.4.0
  Downloading importlib_resources-5.10.1-py3-none-any.whl (34 kB)
Collecting pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0
  Downloading pyrsistent-0.19.2-py3-none-any.whl (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.5/57.5 kB 233.7 kB/s eta 0:00:00
Collecting attrs>=17.4.0
  Downloading attrs-22.1.0-py2.py3-none-any.whl (58 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.8/58.8 kB 136.6 kB/s eta 0:00:00
Collecting pkgutil-resolve-name>=1.3.10
  Downloading pkgutil_resolve_name-1.3.10-py3-none-any.whl (4.7 kB)
Collecting markdown<3.4,>=3.2.1
  Downloading Markdown-3.3.7-py3-none-any.whl (97 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.8/97.8 kB 326.9 kB/s eta 0:00:00
Collecting jinja2>=2.11.1
  Downloading Jinja2-3.1.2-py3-none-any.whl (133 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.1/133.1 kB 222.4 kB/s eta 0:00:00
Collecting packaging>=20.5
  Downloading packaging-22.0-py3-none-any.whl (42 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.6/42.6 kB 313.6 kB/s eta 0:00:00
Collecting watchdog>=2.0
  Downloading watchdog-2.2.0-py3-none-manylinux2014_x86_64.whl (78 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.7/78.7 kB 330.9 kB/s eta 0:00:00
Collecting click>=7.0
  Downloading click-8.1.3-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.6/96.6 kB 333.9 kB/s eta 0:00:00
Collecting mergedeep>=1.3.4
  Using cached mergedeep-1.3.4-py3-none-any.whl (6.4 kB)
Collecting pyyaml-env-tag>=0.1
  Using cached pyyaml_env_tag-0.1-py3-none-any.whl (3.9 kB)
Collecting ghp-import>=1.0
  Downloading ghp_import-2.1.0-py3-none-any.whl (11 kB)
Collecting six>=1.11.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting deprecated
  Using cached Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)
Collecting pynacl>=1.4.0
  Downloading PyNaCl-1.5.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (856 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 856.7/856.7 kB 295.5 kB/s eta 0:00:00
Collecting pyjwt>=2.4.0
  Downloading PyJWT-2.6.0-py3-none-any.whl (20 kB)
Collecting lark-parser<0.8.0,>=0.7.1
  Using cached lark-parser-0.7.8.tar.gz (276 kB)
  Preparing metadata (setup.py) ... done
Collecting pathvalidate<3,>=2.3.0
  Downloading pathvalidate-2.5.2-py3-none-any.whl (20 kB)
Collecting tcolorpy<1,>=0.0.5
  Downloading tcolorpy-0.1.2-py3-none-any.whl (7.9 kB)
Collecting mbstrdecoder<2,>=1.0.0
  Downloading mbstrdecoder-1.1.1-py3-none-any.whl (7.7 kB)
Collecting typepy[datetime]<2,>=1.2.0
  Using cached typepy-1.3.0-py3-none-any.whl (31 kB)
Collecting DataProperty<2,>=0.55.0
  Downloading DataProperty-0.55.0-py3-none-any.whl (26 kB)
Requirement already satisfied: setuptools>=38.3.0 in ./.venv/lib/python3.7/site-packages (from pytablewriter->-r requirements.dev.txt (line 9)) (39.0.1)
Collecting tabledata<2,>=1.3.0
  Using cached tabledata-1.3.0-py3-none-any.whl (11 kB)
Collecting pytest>=4.6
  Downloading pytest-7.2.0-py3-none-any.whl (316 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 316.8/316.8 kB 223.0 kB/s eta 0:00:00
Collecting coverage[toml]>=5.2.1
  Downloading coverage-6.5.0-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (210 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 210.1/210.1 kB 200.5 kB/s eta 0:00:00
Collecting execnet>=1.1
  Downloading execnet-1.9.0-py2.py3-none-any.whl (39 kB)
Collecting requests-toolbelt>=0.9.1
  Downloading requests_toolbelt-0.10.1-py2.py3-none-any.whl (54 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.5/54.5 kB 320.9 kB/s eta 0:00:00
Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.26.13-py2.py3-none-any.whl (140 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 140.6/140.6 kB 184.2 kB/s eta 0:00:00
Collecting charset-normalizer<3,>=2
  Downloading charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Downloading idna-3.4-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.5/61.5 kB 522.5 kB/s eta 0:00:00
Collecting certifi>=2017.4.17
  Downloading certifi-2022.12.7-py3-none-any.whl (155 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 155.3/155.3 kB 415.2 kB/s eta 0:00:00
Collecting argcomplete>=1.8.1
  Downloading argcomplete-2.0.0-py2.py3-none-any.whl (37 kB)
Collecting toml>=0.10.0
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting xmltodict>=0.11.0
  Downloading xmltodict-0.13.0-py2.py3-none-any.whl (10.0 kB)
Collecting zipp>=0.5
  Downloading zipp-3.11.0-py3-none-any.whl (6.6 kB)
Collecting pymdown-extensions>=9.4
  Downloading pymdown_extensions-9.9-py3-none-any.whl (219 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 219.5/219.5 kB 389.2 kB/s eta 0:00:00
Collecting mkdocs-material-extensions>=1.1
  Downloading mkdocs_material_extensions-1.1.1-py3-none-any.whl (7.9 kB)
Collecting pygments>=2.12
  Downloading Pygments-2.13.0-py3-none-any.whl (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 253.0 kB/s eta 0:00:00
Collecting soupsieve>1.2
  Downloading soupsieve-2.3.2.post1-py3-none-any.whl (37 kB)
Collecting dataclasses-json<0.6.0,>=0.5.6
  Downloading dataclasses_json-0.5.7-py3-none-any.whl (25 kB)
Collecting markdown2<3.0.0,>=2.4.1
  Downloading markdown2-2.4.6-py2.py3-none-any.whl (37 kB)
Collecting MarkupSafe<3.0,>=2.0
  Downloading MarkupSafe-2.1.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Collecting pytz<2022.0,>=2021.1
  Using cached pytz-2021.3-py2.py3-none-any.whl (503 kB)
Collecting htmlmin<0.2.0,>=0.1.12
  Using cached htmlmin-0.1.12.tar.gz (19 kB)
  Preparing metadata (setup.py) ... done
Collecting importlib-metadata>=3.10
  Downloading importlib_metadata-4.13.0-py3-none-any.whl (23 kB)
Collecting tomli
  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting marshmallow-enum<2.0.0,>=1.5.1
  Using cached marshmallow_enum-1.5.1-py2.py3-none-any.whl (4.2 kB)
Collecting typing-inspect>=0.4.0
  Downloading typing_inspect-0.8.0-py3-none-any.whl (8.7 kB)
Collecting marshmallow<4.0.0,>=3.3.0
  Downloading marshmallow-3.19.0-py3-none-any.whl (49 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 49.1/49.1 kB 154.0 kB/s eta 0:00:00
Collecting python-dateutil>=2.8.1
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting smmap<6,>=3.0.1
  Using cached smmap-5.0.0-py3-none-any.whl (24 kB)
Collecting chardet<6,>=3.0.4
  Downloading chardet-5.1.0-py3-none-any.whl (199 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 199.1/199.1 kB 300.7 kB/s eta 0:00:00
Collecting cffi>=1.4.1
  Downloading cffi-1.15.1-cp37-cp37m-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (427 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 427.9/427.9 kB 443.2 kB/s eta 0:00:00
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting exceptiongroup>=1.0.0rc8
  Downloading exceptiongroup-1.0.4-py3-none-any.whl (14 kB)
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting wrapt<2,>=1.10
  Downloading wrapt-1.14.1-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (75 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 75.2/75.2 kB 481.5 kB/s eta 0:00:00
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting mypy-extensions>=0.3.0
  Using cached mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
Installing collected packages: pytz, mypy-extensions, multiprocessing_logging, lark-parser, iniconfig, htmlmin, zipp, xmltodict, wrapt, watchdog, urllib3, typing-extensions, tomli, toml, terminaltables, tcolorpy, soupsieve, smmap, six, pyyaml, pyrsistent, pyjwt, pygments, pycparser, pkgutil-resolve-name, pathvalidate, packaging, mkdocs-material-extensions, mergedeep, MarkupSafe, markdown2, idna, giturlparse, execnet, exceptiongroup, coverage, commentjson, charset-normalizer, chardet, certifi, attrs, typing-inspect, requests, pyyaml-env-tag, python-dateutil, pychalk, mbstrdecoder, marshmallow, jinja2, importlib-resources, importlib-metadata, gitdb, deprecated, cffi, beautifulsoup4, webpreview, typepy, requests-toolbelt, pynacl, pluggy, marshmallow-enum, markdown, jsonschema, jsonpickle, gitpython, ghp-import, click, argcomplete, yq, python-gitlab, pytest, pymdown-extensions, pygithub, mkdocs, mdx_truly_sane_lists, dataclasses-json, pytest-xdist, pytest-timeout, pytest-cov, mkdocs-material, json-schema-for-humans, DataProperty, tabledata, pytablewriter
  DEPRECATION: lark-parser is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for lark-parser ... done
  DEPRECATION: htmlmin is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for htmlmin ... done
  DEPRECATION: commentjson is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for commentjson ... done
  DEPRECATION: pychalk is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for pychalk ... done
Successfully installed DataProperty-0.55.0 MarkupSafe-2.1.1 argcomplete-2.0.0 attrs-22.1.0 beautifulsoup4-4.11.1 certifi-2022.12.7 cffi-1.15.1 chardet-5.1.0 charset-normalizer-2.1.1 click-8.1.3 commentjson-0.9.0 coverage-6.5.0 dataclasses-json-0.5.7 deprecated-1.2.13 exceptiongroup-1.0.4 execnet-1.9.0 ghp-import-2.1.0 gitdb-4.0.10 gitpython-3.1.29 giturlparse-0.10.0 htmlmin-0.1.12 idna-3.4 importlib-metadata-4.13.0 importlib-resources-5.10.1 iniconfig-1.1.1 jinja2-3.1.2 json-schema-for-humans-0.44 jsonpickle-3.0.1 jsonschema-4.17.3 lark-parser-0.7.8 markdown-3.3.7 markdown2-2.4.6 marshmallow-3.19.0 marshmallow-enum-1.5.1 mbstrdecoder-1.1.1 mdx_truly_sane_lists-1.3 mergedeep-1.3.4 mkdocs-1.4.2 mkdocs-material-8.5.11 mkdocs-material-extensions-1.1.1 multiprocessing_logging-0.3.3 mypy-extensions-0.4.3 packaging-22.0 pathvalidate-2.5.2 pkgutil-resolve-name-1.3.10 pluggy-1.0.0 pychalk-2.0.1 pycparser-2.21 pygithub-1.57 pygments-2.13.0 pyjwt-2.6.0 pymdown-extensions-9.9 pynacl-1.5.0 pyrsistent-0.19.2 pytablewriter-0.64.2 pytest-7.2.0 pytest-cov-4.0.0 pytest-timeout-2.1.0 pytest-xdist-3.1.0 python-dateutil-2.8.2 python-gitlab-3.12.0 pytz-2021.3 pyyaml-6.0 pyyaml-env-tag-0.1 requests-2.28.1 requests-toolbelt-0.10.1 six-1.16.0 smmap-5.0.0 soupsieve-2.3.2.post1 tabledata-1.3.0 tcolorpy-0.1.2 terminaltables-3.1.10 toml-0.10.2 tomli-2.0.1 typepy-1.3.0 typing-extensions-4.4.0 typing-inspect-0.8.0 urllib3-1.26.13 watchdog-2.2.0 webpreview-1.7.2 wrapt-1.14.1 xmltodict-0.13.0 yq-3.1.0 zipp-3.11.0
```

And finally, run `./build.sh` from MegaLinter root directory

But getting this output:
```
Traceback (most recent call last):
  File "./.automation/build.py", line 20, in <module>
    import megalinter
  File "/home/llaville/devilbox_data/backups/github/megalinter-oxsec/megalinter/__init__.py", line 4, in <module>
    from .MegaLinter import Megalinter
  File "/home/llaville/devilbox_data/backups/github/megalinter-oxsec/megalinter/MegaLinter.py", line 16, in <module>
    from megalinter import (
  File "/home/llaville/devilbox_data/backups/github/megalinter-oxsec/megalinter/flavor_factory.py", line 152, in <module>
    def are_all_repository_linters(linter_names: list[str]) -> bool:
TypeError: 'type' object is not subscriptable
```

What did I missed ?

**Comment by llaville** [comment]
FYI: I try in blind with GitHub Action to read for results.

**Comment by nvuillam** [comment]
I don't use venv for MegaLinter, so probably contributing guide is perfectible :/

**Comment by nvuillam** [comment]
@llaville I merged your branch in my fix branch, and now the only issue is phplint

https://github.com/oxsecurity/megalinter/actions/runs/3720667470/jobs/6310295426

I see you removed the part using phplint.phar, and it seems that for 5.3.0 there is no phplint.phar is released artifacts.... help ? :)

**Comment by nvuillam** [comment]
![image](https://user-images.githubusercontent.com/17500430/208257471-536ddf5e-3ee2-4cd2-8d0b-0376bf56bd07.png)

it seems composer require is not efficient enough to install phplint :/

**Comment by llaville** [comment]
@nvuillam I'll check tomorrow.

**Comment by nvuillam** [comment]
@llaville thanks, that will unlock lots of PRs :)

**Comment by llaville** [comment]
@nvuillam I've forgotten to add the PATH directive in the PHP descriptor file
Add in you patch 
```
diff --git a/megalinter/descriptors/php.megalinter-descriptor.yml b/megalinter/descriptors/php.megalinter-descriptor.yml
index 6c4e66477..acb7f3f1b 100644
--- a/megalinter/descriptors/php.megalinter-descriptor.yml
+++ b/megalinter/descriptors/php.megalinter-descriptor.yml
@@ -152,3 +152,4 @@ linters:
         - |
           RUN composer global require --ignore-platform-reqs overtrue/phplint ^5.3 \
               && composer global config bin-dir --absolute
+        - ENV PATH="/root/.composer/vendor/bin:$PATH"
```
No more needed to use the PHAR archive, that was a temporary solution. We come back to a similar solution applied initially for https://github.com/oxsecurity/megalinter/pull/1037 that require only a Composer global installation !

That should fixed the current issue.

**Comment by llaville** [comment]
Available in v6.16.0 !
